crates/storage/src/alloc/init.rs

```diff
@@ -174,7 +174,11 @@ impl DynamicAllocatorState {
 }
 
 cfg_if! {
-    if #[cfg(all(not(feature = "std"), target_arch = "wasm32"))] {
+    if #[cfg(
+        any(
+            all(not(feature = "std"), target_arch = "wasm32"),
+            features = "ink-experimental-engine"
```

Comment:
this is wrong. this code below is particular for the on-chain environment using SEAL and must not be used with any off-chain environment implementation. Use the same that is used for the current off-chain environment instead.

Reply:
See description of this PR:
>I've integrated engine in a pragmatic way into on_chain. Imo integrating our emulation at such a low layer is a good idea, since testing gets closer to the real on-chain environment. Our current off-chain env is integrated at a much higher layer and while running off-chain tests the whole on-chain layer is not touched in any way, possibly resulting in a different on-chain behavior of a contract.

Best to discuss in person.